nemo/utils/preemption_callback.py

```diff
@@ -0,0 +1,66 @@
+import signal
+import sys
+
+import pytorch_lightning as pl
+import torch
+from pytorch_lightning.callbacks import Callback, ModelCheckpoint
+
+from nemo.collections.common.callbacks.nemomodelcheckpoint import NeMoModelCheckpoint
+
+
+class PreemptionCallback(Callback):
+    def __init__(self, device, checkpoint_callback, sig=signal.SIGTERM):
+        self.sig = sig
+        self.device = device
+        self.checkpoint_callback = checkpoint_callback
```

Comment:
Please add some docstrings for this class and let's add this to the exp_manager docs: https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/core/exp_manager.html

Reply:
Also please add a header to this file

Reply:
@ericharper Added header and docstrings to the class. Should be visible in the most recent version of thePR.